### DISCUSSION TITLE
Performance and disk usage of Milvus when using DiskANN as an Index

### DISCUSSION THREAD

**Discussion by rationalga** [milvus-io/milvus#25467] [Q&A and General discussion]
Hi,
I am running the docker-based default installation of Milvus standalone server(v2.2.11), and I'm using it to index 18 million BERT embeddings (each having 768 dimensions) with a DiskANN index. The configuration for the DiskANN index uses the L2 distance metric and has a search list of 30. The server is running on a machine with 32GB RAM and a 2TB SSD.

I would appreciate if you could comment on the following results.

### Performance
In terms of performance, the total time required for both insertion and index building is 291 minutes. When performing 10,000 queries, the search took 62 seconds (161 QPS).

**Q1.** Given that I'm using a disk-based index, a QPS of 161 seems good enough. Would this be deemed a reasonable rate or could it be considered low?

### Disk Usage
As for disk usage, the size of the Milvus Core directory increases to about 56GB after indexing, while the MinIO directory reaches to 1.2TB. The size of MinIO directory is quite excessive given that the 18 million embeddings I'm storing only amount to around 55GB of data.

**Q2.** Is this disk usage expected? If not, is there an option to disable unneeded log generation, especially considering this is a single node installation where  features like replication and fault tolerance may not be required?

If you think the above results don't makes sense and I may be doing something  wrong then please let me know as well. Thank you very much.

**Comment by yhmo** [comment]
18m 768 dim embeddings, 161 QPS is good for DiskANN index.

DiskANN index on-disk size is about 1.5x~2.0x of the original data size. 1.2TB is beyond expected. I guess there are some middle segments generated and temp index is built. You can observe the size of MinIO directory, does the size decrease slowly?

**Comment by xiaofan-luan** [comment]
this might be a compaction issue. 
If you stop inserting you data might be expired after 1 day.

**Comment by rationalga** [comment]
@yhmo @xiaofan-luan Thank you for your replies. Yes, the disk usage has dropped now. MinIO directory is 116GB instead of 1.2 TB and Milvus directory is 82GB (increased from 56GB). 

During insertion of large dataset (e.g. 20 million embedding), there is a possibility of running out of disk space. Is there a way to the change the expiry of insertion logs to minutes? I mean remove the logs as soon as they are written to buckets (I actually don't fully understand how it works, so I may be very wrong :) ). Your suggestions would be highly appreciated.
Thank you.

**Reply by yhmo** [reply]
In the milvus.yaml, there is a "gc" section under rootcoord:
Set rootcoord.gc.interval to be 600, the garbage collection operations will be triggered with 10 minutes interval.
```
rootcoord:
  gc:
    interval: 3600 # gc interval in seconds
    missingTolerance: 86400 # file meta missing tolerance duration in seconds, 60*24
    dropTolerance: 3600 # file belongs to dropped entity tolerance duration in seconds
```

**Reply by rationalga** [reply]
Thank you very much, i'll try that.
